### PR TITLE
Update to Terraform 0.15

### PR DESCRIPTION
**Pull request by brikis98**
This PR updates the Terratest build to run with Terraform 0.15.

### CODE REVIEW

**Review by zackproser — APPROVED** [review]
LGTM

File: test/azure/terraform_azure_postgresql_example_test.go [hunk 1]
```diff
@@ -1,3 +1,8 @@
+// +build azure
```
Comment (brikis98) [inline test/azure/terraform_azure_postgresql_example_test.go:1]: Missing build tag on our Azure code. Not related to this PR, but was causing test failures.

File: examples/terraform-asg-scp-example/main.tf [hunk 2]
```diff
@@ -26,7 +26,7 @@ provider "aws" {
 resource "aws_launch_template" "sample_launch_template" {
   name_prefix            = var.instance_name
   image_id               = data.aws_ami.ubuntu.id
-  instance_type          = "t2.micro"
+  instance_type          = var.instance_type
```
Comment (brikis98) [inline examples/terraform-asg-scp-example/main.tf:29]: This is not directly related to this PR, but we were getting intermittent test failures when we picked an instance type that happened not to be available in all AZs in an AWS region. To make tests more stable, I made these configurable, and used `GetRecommendedInstanceType` in our test code to pick a type that is available in all AZs.

**Comment by brikis98** [comment]
OK, there is still a bit of test instability, but it's unrelated to this PR, and passed on re-run, and now all tests are passing!

**Review by zackproser — APPROVED** [review]
LGTM

**Comment by brikis98** [comment]
Thanks for the review! Merging now.

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/terratest/releases/tag/v0.33.0

**Pull request merged**